src/erl.rs

```diff
@@ -15,11 +21,10 @@ use heck::SnakeCase;
 use itertools::Itertools;
 use std::char;
 use std::default::Default;
+use std::path::Path;
 use std::sync::Arc;
 
-const INDENT: isize = 4;
-
-pub fn generate_erlang(analysed: &[Analysed]) -> Vec<OutputFile> {
+pub fn generate_erlang(analysed: &[Analysed], project_path: impl AsRef<Path>) -> Vec<OutputFile> {
```

Comment:
This function is being removed soon (possibly the next release) as it's part of the original build system which is rather messy.

The new place to look for code generation is the `codegen` modules. Currently the codegen for record headers and modules are two different code generators, so it'll be difficult to use one from the other like this.

The reason for them being split is that different tools may want the headers in different directories, or not at all in the case of the Gleam compiler up until now.

I think it would be easier to show tuples rather than the record sugar. We will need to implement the tuple syntax for records without labelled fields so we could use them always rather than implement records additionally and then having logic to select between the two.

Reply:
I think I follow this and can come around to resolving it at some point. This relies on records really just being tagged tuples so we can output the description of the tagged tuple directly?